src/Tools/onnx-mlir-opt/onnx-mlir-opt.cpp

```diff
@@ -192,7 +192,12 @@ int main(int argc, char **argv) {
   }
 
   // TODO(imaihal): Change preloadDialectsInContext to false.
-  return failed(mlir::MlirOptMain(output->os(), std::move(file), passPipeline,
-      registry, split_input_file, verify_diagnostics, verify_passes,
-      allowUnregisteredDialects, /*preloadDialectsInContext*/ true));
+  if (failed(mlir::MlirOptMain(output->os(), std::move(file), passPipeline,
+          registry, split_input_file, verify_diagnostics, verify_passes,
+          allowUnregisteredDialects, /*preloadDialectsInContext*/ true))) {
+    return mlir::asMainReturnCode(failure());
+  }
+
+  output->keep();
```

Comment:
Could you add/adapt a test so something will fail if this breaks again?

Reply:
Does it has to be in the onnx repo? Because our end to end tests break if this breaks again

Reply:
Got it into a lit test